server/routes/users.js

```diff
@@ -3,167 +3,473 @@ var express = require('express');
+        res.json(result);
+      } else {
+        util.throwError(404, 'No users matching user_id = ' + req.params.user_id + ' found');
+      }
+    })
+    .catch(err => { return next(err) });
 });
 
 // GET whether user is admin or not
 router.get('/:user_id/admin', function(req, res, next) {
-  const user_id = req.params.user_id;
-  db.query('SELECT is_admin FROM swe_user WHERE id = $1', [user_id], (err, result) => {
-    if (err) return next(err);
-    res.send(result.rows);
-  });
+  knex('swe_user').select('is_admin').where({ id: req.params.user_id })
+    .then(result => {
+      if (result.length) {
+          res.json(result);
+      } else {
+        util.throwError(404, 'No users matching user_id = ' + req.params.user_id + ' found');
+      }
+    })
+    .catch(err => { return next(err) });
 });
 
-// GET a user's past events-HELP NEEDED
+// Update user info
+var inserts = []
+router.put('/:user_id', function(req, res, next) {
+  var user_id = req.params.user_id;
+  var first_name = req.query.first_name;
+  var last_name = req.query.last_name;
+  var password = req.query.password;
+  var email = req.query.email;
+  var phone = req.query.phone;
+  var university_id = req.query.university_id;
+  var is_admin = req.query.is_admin;
+
+  name_values = {
+    first_name: first_name,
+    last_name: last_name
+  }
+  password_value = {
+    password: password
+  }
+  email_value = {
+    email: email
+  }
+  phone_value = {
+    phone: phone
+  } 
+  university_id_value = {
+    university_id: university_id
+  }
+  is_admin_value = {
+    is_admin: is_admin
+  }
+
+  knex.transaction(async function(trx) { 
+    
+    var query1;
+    var query2;
+    var query3;
+    var query4;
+    var query5;
+    if(first_name && last_name){
+      query1 = knex('swe_user').update(name_values).where({id: user_id});
+    }
+    if(password){
+      query2 = knex('swe_user').update(password_value).where({id: user_id});
+    }
+    if(email){
+      query3 = knex('swe_user').update(email_value).where({id: user_id});
+    }
+    if(university_id){
+      query4 = knex('swe_user').update(university_id_value).where({id: user_id});
+    }
+    if(is_admin){
+      query5 = knex('swe_user').update(is_admin_value).where({id: user_id});
+    }
+
+    if(first_name && last_name){
+      await query1.transacting(trx);
+    }
+    if (password){
+      await query2.transacting(trx);
+    }
+    if (email){
+      await query3.transacting(trx);
+    }
+    if (university_id){
+      await query4.transacting(trx);
+    }
+    if (is_admin){
+      await query5.transacting(trx);
+    }
+    return trx.commit;
+  })
+  .then(result => {
+    res.send(util.message('Successfully updated user'));    
+  })
+  .catch(err => { return next(err) 
+    // If we get here, that means that neither the user insert nor any of the other inserts hav taken place
+  })
+});
+
+// GET a user's past events
 router.get('/:user_id/past', function(req, res, next) {
```

Comment:
When indenting, just do this format:

knex('event_checkin')
  .select()
  .innerJoin()
  .etc.

Reply:
fixed, consistent with events api